handlers.go

```diff
@@ -0,0 +1,59 @@
+package dagstore
+
+import (
+	"context"
+
+	"github.com/filecoin-project/dagstore/shard"
+)
+
+// RecoverImmediately takes a failureCh were DAGStore failures are sent, and
+// attempts to recover the shard immediately up until maxAttempts for each
+// unique shard.
+//
+// Attempt tracking does not survive restarts. When the passed context fires,
+// the failure handler goroutine will yield
```

Comment:
> the failure handler goroutine will yield

nit: this tripped me up the first time I read it - there is no go routine :)

Reply:
Good point! I wrote this comment for a previous version ;-)

Reply:
```suggestion
// the failure handler will yield. It is recommended to call this
// method from a dedicated goroutine, as it runs an infinite event
// loop.
```